### PR TITLE
TF-2682 Create update always read receipt interactor and integrate with Settings UI

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2615 
- #2682 

### Demo
**Web**

https://github.com/linagora/tmail-flutter/assets/160106668/9de168f3-2370-492a-a069-3eb6150d5d54

**Mobile**

https://github.com/linagora/tmail-flutter/assets/160106668/08142abe-3819-4422-b11f-76aed2ecf095

### Screenshots
Screenshot 2024-03-08 at 10 17 12
Screenshot 2024-03-08 at 10 17 31
Screenshot 2024-03-08 at 10 17 44
Screenshot 2024-03-08 at 15 27 00
Screenshot 2024-03-08 at 15 27 11
Screenshot 2024-03-08 at 15 27 16

### CODE REVIEW

File: lib/features/manage_account/presentation/always_read_receipt/bindings/always_read_receipt_bindings.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  void bindingsDataSourceImpl() {
+    Get.lazyPut(() => RemoteServerSettingsDataSourceImpl(
+      Get.find<ServerSettingsAPI>(), 
+      Get.find<RemoteExceptionThrower>()));
+  }
+
+  @override
+  void bindingsInteractor() {
+    Get.put(GetAlwaysReadReceiptSettingInteractor(Get.find<ServerSettingsRepository>()));
+    Get.put(UpdateAlwaysReadReceiptSettingInteractor(Get.find<ServerSettingsRepository>()));
```
Comment (dab246) [inline lib/features/manage_account/presentation/always_read_receipt/bindings/always_read_receipt_bindings.dart:36]: why not use `lazyPut `?

File: lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_controller.dart [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  AlwaysReadReceiptController(
+    this._getAlwaysReadReceiptSettingInteractor,
+    this._updateAlwaysReadReceiptSettingInteractor,
+  );
+
+  final GetAlwaysReadReceiptSettingInteractor _getAlwaysReadReceiptSettingInteractor;
+  final UpdateAlwaysReadReceiptSettingInteractor _updateAlwaysReadReceiptSettingInteractor;
+
+  final alwaysReadReceipt = true.obs;
+  final isLoading = false.obs;
```
Comment (dab246) [inline lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_controller.dart:23]: Why use the variable `Obx`?

File: lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_view.dart [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                          }),
+                        ),
+                      ),
+                      const SizedBox(width: 12),
+                      Expanded(
+                        child: Text(
+                          AppLocalizations.of(context).emailReadReceiptsToggleDescription,
+                          style: const TextStyle(
+                            fontSize: 16,
+                            height: 20 / 16,
```
Comment (dab246) [inline lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_view.dart:66]: why is that so `20 / 16`?

Comment (tddang-linagora) [reply lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_view.dart:66]: Line height of the text here is 20. In Flutter, a text can be set line height to `x`, when it has font size `y`, with `height: x / y`. [See the docs](https://api.flutter.dev/flutter/painting/TextStyle/height.html)

File: lib/features/manage_account/presentation/always_read_receipt/bindings/always_read_receipt_bindings.dart (hunk 1 above)
Comment (tddang-linagora) [reply lib/features/manage_account/presentation/always_read_receipt/bindings/always_read_receipt_bindings.dart]: Done

File: lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_view.dart [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import 'package:core/core.dart';
```
[the remark this answers is not in the document]
Comment (tddang-linagora) [reply lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_view.dart]: Done

**Review by dab246 — APPROVED** [review]
please rebase it

**Comment by hoangdat** [comment]
Please change the loading to the top

image

You can refer to the `email recovery` or `mark as read` function. 
image

**Comment by tddang-linagora** [comment]
> Please change the loading to the top
> image


> Please change the loading to the top
> image
> 
> You can refer to the `email recovery` or `mark as read` function. image

Done

File: lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_controller.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (tddang-linagora) [reply lib/features/manage_account/presentation/always_read_receipt/always_read_receipt_controller.dart]: Done

**Review by hoangdat — APPROVED** [review]

**Pull request merged**